### PR TITLE
Add new shortcodes for Giphy and Vine

### PR DESCRIPTION
**Pull request by danielbachhuber**
See #83, #81

### CODE REVIEW

File: inc/shortcodes/class-giphy.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+	public static function reversal( $content ) {
+
+		$needle = '#<iframe[^>]+src="//giphy\.com/embed/([\w]+)"[^>]+></iframe>(<p><a[^>]+>via GIPHY</a></p>)?#';
+		if ( preg_match_all( $needle, $content, $matches ) ) {
+			$replacements = array();
+			$shortcode_tag = self::get_shortcode_tag();
+			foreach ( $matches[0] as $key => $value ) {
+				$replacement_url = 'http://giphy.com/gifs/' . $matches[1][ $key ];
+				$replacements[ $value ] = '[' . $shortcode_tag . ' url="' . esc_url( $replacement_url ) . '"]';
```
Comment (goldenapples) [inline inc/shortcodes/class-giphy.php]: Any reason we're storing the full URL rather than just the gif ID?

File: inc/shortcodes/class-vine.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	public static function reversal( $content ) {
+
+		$needle = '#<iframe[^>]+src="https?://vine\.co/v/([\w]+)(/[^"]+)?"[^>]+></iframe>#';
+		if ( preg_match_all( $needle, $content, $matches ) ) {
+			$replacements = array();
+			$shortcode_tag = self::get_shortcode_tag();
+			foreach ( $matches[0] as $key => $value ) {
+				$replacement_url = 'https://vine.co/v/' . $matches[1][ $key ];
```
Comment (goldenapples) [inline inc/shortcodes/class-vine.php]: Same question here. If that's a pattern established elsewhere, I'm fine with it. Just curious what the rationale was..

File: inc/shortcodes/class-giphy.php (hunk 1 above)
Comment (danielbachhuber) [inline inc/shortcodes/class-giphy.php]: Usability — easier to copy and paste a URL into a field, than have to identify and enter the ID.

Comment (goldenapples) [inline inc/shortcodes/class-giphy.php]: Also, if we do store full URLs, seems like the https link would be preferable.

Comment (goldenapples) [inline inc/shortcodes/class-giphy.php]: eh, never mind that last comment. Looks like giphy's direct https links don't really work.

**Pull request merged**